src/audio/drc/drc_math_generic.c

```diff
@@ -106,15 +106,16 @@ inline int32_t drc_log_fixed(int32_t x)
 
 /*
  * Input is Q2.30: (-2.0, 2.0)
- * Output range: (-1.0, 1.0); regulated to Q1.31: (-1.0, 1.0)
+ * Output range: (-1.0, 1.0); regulated to Q1.15: (-1.0, 1.0)
  */
 inline int32_t drc_sin_fixed(int32_t x)
 {
 	const int32_t PI_OVER_TWO = Q_CONVERT_FLOAT(1.57079632679489661923f, 30);
+	uint32_t flags = EN_CORDIC_SINE;
+	/* input range of sin_cos_fixed_16b() is non-negative */
+	int16_t abs_sin_val = sin_cos_fixed_16b(q_mult(ABS(x), PI_OVER_TWO, 30, 30, 28), flags);
 
-	/* input range of sin_fixed() is non-negative */
-	int32_t abs_sin_val = sin_fixed(q_mult(ABS(x), PI_OVER_TWO, 30, 30, 28));
-	return SGN(x) < 0 ? -abs_sin_val : abs_sin_val;
+	return x < 0 ? -abs_sin_val : abs_sin_val;
```

Comment:
it's int16_t abs_sin_val not int32_t abs_sin_val

Reply:
You could do the shift to Q1.31 compatible format in return statement . Since the abs_sin_value is positive it won't overflow when converted to negative.
`return x < 0 ? -abs_sin_val << 16: abs_sin_val << 16`

Reply:
we need to use "int32_t abs_sin_val" here otherwise we are not able to do the left shift.